### PR TITLE
Feature: Added posibility to pass priority for the message at write time

### PR DESCRIPTION
**Pull request by xytis**
I am writing syslog forwarder using this library and I need to pass priority at write moment.

### CODE REVIEW

File: writer.go [hunk 1]
```diff
@@ -59,6 +59,11 @@ func (w *Writer) Write(b []byte) (int, error) {
 	return w.writeAndRetry(w.priority, string(b))
 }
 
+// RawWrite sends a log message with a custom priority
+func (w *Writer) RawWrite(p Priority, b []byte) (int, error) {
```
Comment (sirsean) [inline writer.go]: `RawWrite` isn't the best name for this. Perhaps `WriteWithPriority` or something would be better.

Why does this take a `[]byte` when all the other write functions take a `string` (and this has to convert to a `string` anyway)?

Comment (xytis) [inline writer.go]: I've tried to be similar to Write method, for me it does not matter what kind of data I pass to the method. But []byte matches syslog message struct.
I'll rename the method.

Comment (xytis) [inline writer.go]: Regarding test coverage, Write method is not covered either. Could you suggest what should I test in this case?

I am not that good at go, so I have trouble expressing my tests. At most what I could think of testing is "does Write produce a log message with default priority" and "does <my new write> produce a log message with given priority". Yet I could not understand existing test cases to implement my tests using code from there. Alternatively to my first question, you could simply help me to understand existing cases.

Comment (sirsean) [inline writer.go]: Take a look at the `TestDebug`, `TestInfo`, etc functions in `writer_test.go`.

Those ones set up a writer with a default priority and then verify that they're able to send log messages with a priority different from that default. That's pretty much exactly what you want here.

So if you had a `TestWriteWithPriority` function that was structured very similarly to `TestDebug`, that would probably get you most of the way there.

**Comment by sirsean** [comment]
Looks good, thanks for the contribution. 👍

**Pull request merged**